crates/forge/tests/integration/syscalls.rs

```diff
@@ -146,7 +147,193 @@ fn test_call_syscall_fail_in_test_fn() {
+    "#
+    ));
+
+    let result = run_test_case(&test);
+
+    assert_case_output_contains!(
+        result,
+        "test_execute_cairo_keccak",
+        "Invalid keccak input size"
+    );
+
+    assert_failed!(result);
+}
+
+#[test]
+fn test_cairo_keccak() {
+    let test = test_case!(indoc!(
+        r#"
+        use array::ArrayTrait;
+        use keccak::cairo_keccak;
+
+        #[test]
+        fn test_execute_cairo_keccak() {
+            let mut input = array![
+                0x0000000000000001,
+                0x0000000000000002,
+                0x0000000000000003,
+                0x0000000000000004,
+                0x0000000000000005,
+                0x0000000000000006,
+                0x0000000000000007,
+                0x0000000000000008,
+                0x0000000000000009,
+                0x000000000000000a,
+                0x000000000000000b,
+                0x000000000000000c,
+                0x000000000000000d,
+                0x000000000000000e,
+                0x000000000000000f,
+                0x0000000000000010,
+                0x0000000000000011
+            ];
+
+            let res = keccak::cairo_keccak(ref input, 0, 0);
+
+            assert(@res.low == @0x5d291eebae35b254ff50ec1fc57832e8, 'Wrong hash low');
+            assert(@res.high == @0x210740d45b1fe2ac908a497ef45509f5, 'Wrong hash high');
+        }
+    "#
+    ));
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+}
+
+#[test]
+fn test_keccak_syscall_in_contract() {
+    let test = test_case!(
+        indoc!(
+            r#"
+            use result::ResultTrait;
+            use array::ArrayTrait;
+            use option::OptionTrait;
+            use traits::TryInto;
+            use starknet::ContractAddress;
+            use snforge_std::{ declare, ContractClassTrait };
+
+            #[starknet::interface]
+            trait IHelloKeccak<TContractState> {
+                fn run_keccak(ref self: TContractState, input: Array<u64>) -> u256;
+            }
+
+            #[test]
+            fn test_keccak_simple() {
+                let contract = declare('HelloKeccak');
+                let contract_address = contract.deploy(@ArrayTrait::new()).unwrap();
+                let dispatcher = IHelloKeccakDispatcher { contract_address };
+
+                let res = dispatcher.run_keccak(array![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17]);
+                assert(
+                    res == u256 { low: 0xec687be9c50d2218388da73622e8fdd5, high: 0xd2eb808dfba4703c528d145dfe6571af },
+                    'Wrong hash value'
+                );
+            }
+        "#
+        ),
+        Contract::from_code_path(
+            "HelloKeccak".to_string(),
+            Path::new("tests/data/contracts/keccak_usage.cairo"),
+        )
+        .unwrap()
+    );
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+}
+
+#[test]
+fn compare_keccak_from_contract_with_plain_keccak() {
```

Comment:
So I assume blockifier keccak == starknet keccak != plain cairo keccak?

Reply:
They all are the same. We just want to check if there are no unexpected behaviours when calling it from the contract